### PR TITLE
Use use Java 7 instead of custom logic for relative paths, issue #857

### PR DESCRIPTION
**Pull request by mkordas**

### CODE REVIEW

**Comment by romani** [comment]
Postponed till 6.6

**Comment by damianszczepanik** [comment]
Looks like you need to rebase after my PR :( but with #861 we may have Utils class fully covered :)

File: pom.xml [hunk 1]
```diff
@@ -588,7 +588,6 @@
           <regex><pattern>.*.PropertiesExpander</pattern><branchRate>50</branchRate><lineRate>83</lineRate></regex>
           <regex><pattern>.*.PropertyCacheFile</pattern><branchRate>22</branchRate><lineRate>19</lineRate></regex>
           <regex><pattern>.*.TreeWalker</pattern><branchRate>90</branchRate><lineRate>90</lineRate></regex>
-          <regex><pattern>com.puppycrawl.tools.checkstyle.Utils</pattern><branchRate>84</branchRate><lineRate>91</lineRate></regex>
```
Comment (mkordas) [inline pom.xml]: :100:

**Comment by romani** [comment]
1) 
> public static String relativeNormalizedPath(final String baseDirectory, final String path)

please name methods in verb notation, Java is not functional language, please do not mix naming notations.

2) 
> CheckerTest.java

I expect that such test (with the same weird inputs path) should appear in Utils test class. As right now whole responsibility on normalization is on Util class.
You can extend tests , but you should not remove old test cases.

**Comment by mkordas** [comment]
1) Done
2) I've already added `testRelativeNormalizedPathWithDenormalizedBaseDirectory` which checks the same thing in simpler way, as `PATH_DENORMALIZER` is extracted to constant. There is just one slight difference - I cannot use denormalizer with backslashes, as backslashes are valid filename characters in Unix. So the previous algorithm had a bug. I hope by using Java 7 util we won't have such bugs anymore.

**Comment by romani** [comment]
merged as FF

**Pull request closed without merging**